libs/sensors/src/PollingSensors.cpp

```diff
@@ -0,0 +1,33 @@
+#include "PollingSensors.h"
+
+#include "Logger.h"
+
+using namespace Sensor;
+
+PollingSensors::PollingSensors(PinName moisture_in, PinName co2_in)
+    : moisture_in_adc(moisture_in), CO2_in_adc(co2_in) {}
+
+// The function currently logs a status message. Can be changed to return just a float indicating the sensor value
+float PollingSensors::moisture_monitoring() {
+  float moisture_reading = (moisture_in_adc.read_voltage() * 5.0) / 1023.0;
```

Comment:
How did you get this equation?

Reply:
So the in the datasheet, they just perform a Arduino analogRead. So this equation is a mbed equivalent for that. But now I just changed it to better reflect the moisture reading (operating voltages - 0 -4.2V and range 0-950).